### PR TITLE
Add email forwarding channel support for inbound email imports

### PR DESCRIPTION
**Pull request by FelixMalfait**
## Summary
This PR implements email forwarding channel functionality, allowing users to create dedicated email addresses that forward messages into Twenty. The system polls an S3 bucket for incoming emails, parses them, matches them to forwarding channels, and imports them as messages.

### Backend Infrastructure
- **Email Forwarding Channel Type**: Added `EMAIL_FORWARDING` as a new `MessageChannelType` and `ConnectedAccountProvider`
- **S3-based Message Queue**: Implemented a bucket-as-queue pattern using S3 prefixes (`incoming/`, `processed/`, `unmatched/`, `failed/`) for reliable at-least-once message processing
- **Inbound Email Parser Service**: Parses raw MIME messages using PostalMime, extracts metadata (sender, recipients, subject, attachments), and detects loopback emails via `X-Twenty-Origin` header
- **Envelope Recipient Extraction**: Recovers the SMTP envelope recipient from email headers (Received, Delivered-To, X-Original-To) with fallback to To/Cc fields
- **Import Orchestration**: `InboundEmailImportService` coordinates the full import cycle: S3 retrieval → parsing → channel lookup → message persistence
- **Polling Cron Job**: Minute-based cron job that lists incoming S3 objects and enqueues import jobs via BullMQ
- **Configuration**: Added `INBOUND_EMAIL_DOMAIN`, `INBOUND_EMAIL_S3_BUCKET`, and `INBOUND_EMAIL_S3_REGION` config variables

### Frontend
- **Email Forwarding Modal**: New modal component displaying the generated forwarding address with copy-to-clipboard functionality
- **Channel Creation Hook**: `useCreateEmailForwardingChannel` mutation hook for creating forwarding channels
- **Settings Integration**: Updated empty state card to include email forwarding channel creation option

### Database
- Migration to add `EMAIL_FORWARDING` enum value to `messageChannel_type_enum`

### Message Outbound Integration
- Added `originWorkspaceId` field to `SendMessageInput` to enable loopback detection
- Modified mail composer to include `X-Twenty-Origin` header when sending from forwarding channels

## Notable Implementation Details
- **Loopback Prevention**: Emails originating from the same workspace are dropped to prevent infinite loops
- **Lazy S3 Client**: S3 client is constructed on first use, allowing clean server boot in workspaces without email forwarding enabled
- **Comprehensive Error Handling**: Import outcomes distinguish between unconfigured, unmatched, parse failures, and persistence failures
- **Extensive Test Coverage**: Full test suite for import service, parser service, and envelope recipient extraction with edge cases
- **Atomic Operations**: S3 moves are used as acknowledgments, ensuring messages aren't lost on retry

https://claude.ai/code/session_01KpyF6p4cUEnuaT4h8DP5Pm

### CODE REVIEW

**Comment by FelixMalfait** [comment]
Ignore!

File: packages/twenty-front/src/modules/settings/accounts/components/SettingsAccountsEmailForwardingModal.tsx [hunk 1]
```diff
@@ -0,0 +1,128 @@
+import { ModalStatefulWrapper } from '@/ui/layout/modal/components/ModalStatefulWrapper';
+import { useModal } from '@/ui/layout/modal/hooks/useModal';
+import { styled } from '@linaria/react';
+import { useLingui } from '@lingui/react/macro';
+import { IconCopy, IconMail } from 'twenty-ui/display';
+import { Button } from 'twenty-ui/input';
+import { themeCssVariables } from 'twenty-ui/theme-constants';
+import { useCopyToClipboard } from '~/hooks/useCopyToClipboard';
+
+export const EMAIL_FORWARDING_MODAL_ID = 'email-forwarding-address-modal';
```
Comment (FelixMalfait) [inline packages/twenty-front/src/modules/settings/accounts/components/SettingsAccountsEmailForwardingModal.tsx]: One export = One file always (check all PR)

File: packages/twenty-front/src/modules/settings/accounts/components/SettingsAccountsListEmptyStateCard.tsx [hunk 2]
```diff
@@ -45,34 +60,69 @@ export const SettingsAccountsListEmptyStateCard = () => {
...
+          />
+        )}
+
+        {isImapSmtpCaldavEnabled && (
+          <UndecoratedLink
+            to={getSettingsPath(SettingsPath.NewImapSmtpCaldavConnection)}
+          >
+            <SettingsCard
+              Icon={<IconAt size={theme.icon.size.md} />}
+              title={t`Connect Account`}
```
Comment (FelixMalfait) [inline packages/twenty-front/src/modules/settings/accounts/components/SettingsAccountsListEmptyStateCard.tsx]: I think we need to rename that to make the distinction more clear with what we just introduce, let's mention IMAP/SMTP/Caldav directly probably?

File: packages/twenty-front/src/modules/settings/accounts/components/SettingsAccountsListEmptyStateCard.tsx [hunk 3]
```diff
@@ -45,34 +60,69 @@ export const SettingsAccountsListEmptyStateCard = () => {
...
+        )}
 
-      {(isMicrosoftMessagingEnabled || isMicrosoftCalendarEnabled) && (
         <SettingsCard
-          Icon={<IconMicrosoft size={theme.icon.size.md} />}
-          title={t`Connect with Microsoft`}
-          onClick={() => triggerApisOAuth(ConnectedAccountProvider.MICROSOFT)}
+          Icon={<IconMail size={theme.icon.size.md} />}
+          title={t`Add Email Forwarding`}
+          description={t`Receive emails from a shared mailbox or Google Group`}
```
Comment (FelixMalfait) [inline packages/twenty-front/src/modules/settings/accounts/components/SettingsAccountsListEmptyStateCard.tsx]: I don't think having a description will look nice since the others don't

File: packages/twenty-front/src/modules/settings/accounts/components/SettingsAccountsListEmptyStateCard.tsx [hunk 4]
```diff
@@ -45,34 +60,69 @@ export const SettingsAccountsListEmptyStateCard = () => {
     isImapSmtpCaldavEnabledState,
   );
 
+  const handleCreateEmailForwardingChannel = useCallback(async () => {
```
Comment (FelixMalfait) [inline packages/twenty-front/src/modules/settings/accounts/components/SettingsAccountsListEmptyStateCard.tsx:61]: I'm a bit surprised to see the logic for email forwarding here and not on a dedicated page? This feels odd

File: packages/twenty-front/src/modules/settings/accounts/graphql/mutations/createEmailForwardingChannel.ts [hunk 5]
```diff
@@ -0,0 +1,18 @@
+import { gql } from '@apollo/client';
+
+export const CREATE_EMAIL_FORWARDING_CHANNEL = gql`
+  mutation CreateEmailForwardingChannel {
+    createEmailForwardingChannel {
+      messageChannel {
+        id
+        handle
+        visibility
+        type
```
Comment (FelixMalfait) [inline packages/twenty-front/src/modules/settings/accounts/graphql/mutations/createEmailForwardingChannel.ts:10]: Curious what type is here?

File: packages/twenty-server/src/engine/core-modules/twenty-config/config-variables.ts [hunk 6]
```diff
@@ -1622,6 +1622,43 @@ export class ConfigVariables {
...
+  INBOUND_EMAIL_DOMAIN: string;
+
+  @ConfigVariablesMetadata({
+    group: ConfigVariablesGroup.AWS_SES_SETTINGS,
+    description:
+      'S3 bucket where SES inbound action writes raw messages. Required to enable email forwarding channels.',
+    type: ConfigVariableType.STRING,
+  })
+  @IsOptional()
+  INBOUND_EMAIL_S3_BUCKET: string;
```
Comment (FelixMalfait) [inline packages/twenty-server/src/engine/core-modules/twenty-config/config-variables.ts]: Why use a new bucket and not a folder within our existing bucket? It seems it adds complexity, especially for self-hosters? Let's find a way to do it SECURLY but in the same bucket

File: packages/twenty-server/src/engine/metadata-modules/message-channel/message-channel-metadata.service.ts [hunk 7]
```diff
@@ -172,6 +186,92 @@ export class MessageChannelMetadataService {
...
+      !isNonEmptyString(inboundEmailBucket) ||
+      !isNonEmptyString(inboundEmailRegion)
+    ) {
+      throw new MessageChannelException(
+        'Email forwarding is not configured: INBOUND_EMAIL_DOMAIN, INBOUND_EMAIL_S3_BUCKET, and a region (INBOUND_EMAIL_S3_REGION or AWS_SES_REGION) must all be set',
+        MessageChannelExceptionCode.EMAIL_FORWARDING_NOT_CONFIGURED,
+      );
+    }
+
+    const existingChannel = await this.repository.findOne({
```
Comment (FelixMalfait) [inline packages/twenty-server/src/engine/metadata-modules/message-channel/message-channel-metadata.service.ts]: I'm not sure I understand what we're trying to do here? There could be multiple groups connected (contactc@mycompany.com ; support@mycompany.com ; etc). That's also one of the reasons we need an intermediate page because we need to collect that handle!

File: packages/twenty-server/src/modules/messaging/message-import-manager/drivers/inbound-email/constants/inbound-email.constants.ts [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+// removes it from the queue, which gives us at-least-once semantics without
+// needing a separate dedupe store.
+export const INBOUND_EMAIL_S3_PREFIXES = {
+  incoming: 'incoming/',
+  processed: 'processed/',
+  unmatched: 'unmatched/',
+  failed: 'failed/',
+} as const;
+
+// Custom header stamped on outbound mail so we can drop echoes when group
```
Comment (FelixMalfait) [inline packages/twenty-server/src/modules/messaging/message-import-manager/drivers/inbound-email/constants/inbound-email.constants.ts]: 1 export = 1 file

File: packages/twenty-server/src/modules/messaging/message-import-manager/drivers/inbound-email/providers/inbound-email-s3-client.provider.ts [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+import { isNonEmptyString } from '@sniptt/guards';
+
+import { TwentyConfigService } from 'src/engine/core-modules/twenty-config/twenty-config.service';
+
+// Lazy S3 client: constructed on first call so the server boots cleanly in
+// workspaces that don't enable email forwarding. Credentials are reused from
+// the AWS_SES_* config block — SES inbound action already writes to this
+// bucket, so the same principal can read it.
+@Injectable()
+export class InboundEmailS3ClientProvider {
```
Comment (FelixMalfait) [inline packages/twenty-server/src/modules/messaging/message-import-manager/drivers/inbound-email/providers/inbound-email-s3-client.provider.ts:10]: I don't think this is really needed. We should use the same bucket and S3 client we already have?

File: packages/twenty-server/src/modules/messaging/message-import-manager/drivers/inbound-email/services/inbound-email-import.service.ts [hunk 10]
```diff
@@ -0,0 +1,199 @@
...
+import { MessageChannelEntity } from 'src/engine/metadata-modules/message-channel/entities/message-channel.entity';
+import { GlobalWorkspaceOrmManager } from 'src/engine/twenty-orm/global-workspace-datasource/global-workspace-orm.manager';
+import { buildSystemAuthContext } from 'src/engine/twenty-orm/utils/build-system-auth-context.util';
+import { InboundEmailParserService } from 'src/modules/messaging/message-import-manager/drivers/inbound-email/services/inbound-email-parser.service';
+import { InboundEmailS3ClientProvider } from 'src/modules/messaging/message-import-manager/drivers/inbound-email/providers/inbound-email-s3-client.provider';
+import { InboundEmailStorageService } from 'src/modules/messaging/message-import-manager/drivers/inbound-email/services/inbound-email-storage.service';
+import { extractEnvelopeRecipient } from 'src/modules/messaging/message-import-manager/drivers/inbound-email/utils/extract-envelope-recipient.util';
+import { MessagingSaveMessagesAndEnqueueContactCreationService } from 'src/modules/messaging/message-import-manager/services/messaging-save-messages-and-enqueue-contact-creation.service';
+
+export type InboundEmailImportOutcome =
```
Comment (FelixMalfait) [inline packages/twenty-server/src/modules/messaging/message-import-manager/drivers/inbound-email/services/inbound-email-import.service.ts]: One export = 1 file

File: packages/twenty-server/src/modules/messaging/message-import-manager/drivers/inbound-email/services/inbound-email-parser.service.ts [hunk 11]
```diff
@@ -0,0 +1,139 @@
...
+import { MessageDirection } from 'src/modules/messaging/common/enums/message-direction.enum';
+import { MessageParticipantRole } from 'twenty-shared/types';
+
+import { type EmailAddress } from 'src/modules/messaging/message-import-manager/types/email-address';
+import { type MessageWithParticipants } from 'src/modules/messaging/message-import-manager/types/message';
+import { formatAddressObjectAsParticipants } from 'src/modules/messaging/message-import-manager/utils/format-address-object-as-participants.util';
+import { sanitizeString } from 'src/modules/messaging/message-import-manager/utils/sanitize-string.util';
+import { X_TWENTY_ORIGIN_HEADER } from 'src/modules/messaging/message-import-manager/drivers/inbound-email/constants/inbound-email.constants';
+
+export type ParsedInboundMessage = {
```
Comment (FelixMalfait) [inline packages/twenty-server/src/modules/messaging/message-import-manager/drivers/inbound-email/services/inbound-email-parser.service.ts]: 1 export 1 file

File: packages/twenty-server/src/modules/messaging/message-import-manager/drivers/inbound-email/services/inbound-email-parser.service.ts [hunk 12]
```diff
@@ -0,0 +1,139 @@
...
+    return {
+      // The S3 key is the natural external id: it uniquely identifies the
+      // object that produced this message and survives retries.
+      externalId: `inbound-email:${s3Key}`,
+      messageThreadExternalId: this.extractThreadId(parsed),
+      headerMessageId: parsed.messageId?.trim() || `inbound-${s3Key}`,
+      subject: sanitizeString(parsed.subject || ''),
+      text: sanitizeString(parsed.text || ''),
+      receivedAt: parsed.date ? new Date(parsed.date) : new Date(),
+      // Forwarding channels are inbound-only — every message is incoming.
```
Comment (FelixMalfait) [inline packages/twenty-server/src/modules/messaging/message-import-manager/drivers/inbound-email/services/inbound-email-parser.service.ts]: No need for useless/obvious comments like this one

File: packages/twenty-server/src/modules/messaging/message-import-manager/drivers/inbound-email/services/inbound-email-parser.service.ts [hunk 13]
```diff
@@ -0,0 +1,139 @@
...
+      direction: MessageDirection.INCOMING,
+      attachments: (parsed.attachments || []).map((attachment) => ({
```
Comment (FelixMalfait) [inline packages/twenty-server/src/modules/messaging/message-import-manager/drivers/inbound-email/services/inbound-email-parser.service.ts]: We don't manage attachments for inbound emails I think?

File: packages/twenty-server/src/modules/messaging/message-import-manager/drivers/inbound-email/services/inbound-email-parser.service.ts [hunk 14]
```diff
@@ -0,0 +1,139 @@
...
+        filename: attachment.filename || 'unnamed-attachment',
+      })),
+      participants: this.extractParticipants(parsed),
+    };
+  }
+
+  private extractThreadId(parsed: ParsedEmail): string {
```
Comment (FelixMalfait) [inline packages/twenty-server/src/modules/messaging/message-import-manager/drivers/inbound-email/services/inbound-email-parser.service.ts]: I wonder if some of these utils have already been built for the IMAP driver and should be shared? We want to avoid duplicated code

File: packages/twenty-server/src/modules/messaging/message-import-manager/drivers/inbound-email/services/inbound-email-storage.service.ts [hunk 15]
```diff
@@ -0,0 +1,126 @@
...
+
+import { INBOUND_EMAIL_S3_PREFIXES } from 'src/modules/messaging/message-import-manager/drivers/inbound-email/constants/inbound-email.constants';
+import { InboundEmailS3ClientProvider } from 'src/modules/messaging/message-import-manager/drivers/inbound-email/providers/inbound-email-s3-client.provider';
+
+// Thin wrapper around S3 that implements the "bucket-as-queue" pattern:
+// listIncoming => pull work, getRawMessage => download, moveTo* => ack.
+// Archive destinations are date-partitioned so operators can browse the
+// bucket by day when triaging failures.
+@Injectable()
+export class InboundEmailStorageService {
```
Comment (FelixMalfait) [inline packages/twenty-server/src/modules/messaging/message-import-manager/drivers/inbound-email/services/inbound-email-storage.service.ts:15]: Let's step back and think about the security model here. Do we have for example one folder per workspace once it's been triaged? I think it could be nice to try and mitigate security risks. Let's see our options

File: packages/twenty-server/src/modules/messaging/message-import-manager/drivers/inbound-email/utils/extract-envelope-recipient.util.ts [hunk 16]
```diff
@@ -0,0 +1,127 @@
+import { type Email as ParsedEmail } from 'postal-mime';
+
+// SES inbound action writes the raw MIME to S3 without preserving the SMTP
+// envelope ("RCPT TO"). We recover the envelope recipient by walking the
+// headers in priority order: delivery headers first (most reliable), then
+// To/Cc as a fallback. The first address whose domain matches our inbound
+// domain wins, which is what we use to look up the forwarding channel.
+
```
Comment (FelixMalfait) [inline packages/twenty-server/src/modules/messaging/message-import-manager/drivers/inbound-email/utils/extract-envelope-recipient.util.ts:8]: Don't we have similar utils from IMAP already? Let's share if we do 

Also let's weight using a library if it's something complex with lots of edge cases? We don't love lib but they can make sense

File: packages/twenty-server/src/modules/messaging/message-import-manager/jobs/messaging-inbound-email-import.job.ts [hunk 17]
```diff
@@ -0,0 +1,28 @@
...
+import { MessageQueue } from 'src/engine/core-modules/message-queue/message-queue.constants';
+import { InboundEmailImportService } from 'src/modules/messaging/message-import-manager/drivers/inbound-email/services/inbound-email-import.service';
+
+export type MessagingInboundEmailImportJobData = {
+  s3Key: string;
+};
+
+// One job per S3 object. Keeping the unit of work small means a single bad
+// message can be retried or moved to `failed/` without blocking the rest
+// of the queue, and lets BullMQ do the per-message dedupe/backoff we want.
```
Comment (FelixMalfait) [inline packages/twenty-server/src/modules/messaging/message-import-manager/jobs/messaging-inbound-email-import.job.ts]: No need for comments like that
